src/sources.rs

```diff
@@ -89,19 +93,30 @@ impl Iter {
     fn visit_next_directory(&mut self, dir_entry: Entry) -> io::Result<()> {
         let readdir = try!(fs::read_dir(&dir_entry.path));
         self.report.increment("source.visited.directories", 1);
-        let mut children = Vec::<(OsString, bool)>::new();
+        let mut children = Vec::<(OsString, bool, Apath)>::new();
         for entry in readdir {
             let entry = try!(entry);
             let ft = try!(entry.file_type());
-            children.push((entry.file_name(), ft.is_dir()));
+            let mut path = dir_entry.apath.to_string().clone();
```

Comment:
You shouldn't need the `.clone()`? Generally `.to_anything` returns a new object (https://aturon.github.io/style/naming.html) and it then shouldn't need to be cloned.

Reply:
Can't get borrow the apath mutable without cloning because of the `to_string` method returns an reference: [to_string() in apath.rs](https://github.com/sourcefrog/conserve/blob/master/src/apath.rs#L33)